app/src/main/java/com/project/myapp/screens/auth/AuthViewModel.kt

```diff
@@ -0,0 +1,121 @@
+package com.project.myapp.screens.auth
+
+import android.util.Patterns
+import androidx.core.text.isDigitsOnly
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+/**
+ * ViewModel for saving AuthActivity state
+ */
+
+class AuthViewModel : ViewModel() {
+    /**
+     * Variables to track the states
+     */
+    private val _credentialsState = MutableStateFlow<AuthState>(AuthState.Initial)
+    val credentialsState: StateFlow<AuthState> get() = _credentialsState
+
+    private val _emailState = MutableStateFlow<AuthState.EmailState>(AuthState.EmailState.Initial)
+    val emailState: StateFlow<AuthState.EmailState> get() = _emailState
+
+    private val _passwordState = MutableStateFlow<AuthState.PasswordState>(AuthState.PasswordState.Initial)
+    val passwordState: StateFlow<AuthState.PasswordState> get() = _passwordState
+
+    fun validateEmail(email: String) {
+        val pattern = Patterns.EMAIL_ADDRESS.matcher(email).matches()
+        if (!pattern && email.isNotEmpty()) {
+            _emailState.value = AuthState.EmailState.Error
+        } else if (email.isEmpty()) {
+            _emailState.value = AuthState.EmailState.ErrorEmpty
+        } else if (pattern) {
+            _emailState.value = AuthState.EmailState.Valid
+        }
```

Comment:
what do you think about this?
```suggestion
      _emailState.value = if (email.isEmpty()) {
             AuthState.EmailState.ErrorEmpty
        } else if (!pattern) {
             AuthState.EmailState.Error
        } else {
             AuthState.EmailState.Valid
        }
```

Reply:
Your solution is definitely better and clearer. I changed [it](https://github.com/cValentyna/myAndroidProject/pull/11/commits/08ab598199580641ae3d9cca49db7081f02819a0)